os/arch/arm/src/amebad/amebad_spi.c

```diff
@@ -0,0 +1,1229 @@
+ *   frequency - The SPI frequency requested
+ *
+ * Returned Value:
+ *   Returns the actual frequency selected
+ *
+ ************************************************************************************/
+
+static uint32_t amebad_spi_setfrequency(FAR struct spi_dev_s *dev,
+					uint32_t frequency)
+{
+	FAR struct amebad_spidev_s *priv = (FAR struct amebad_spidev_s *)dev;
+
+	/* Has the SPI bus frequency changed? */
+
+	if (frequency != priv->frequency) {
+		/* Disable SPI if it is enabled */
+
+		priv->frequency = frequency;
+
+		if (priv->spi_idx == MBED_SPI0) 
+			spi_frequency(priv->spi_object, priv->frequency);
+	}
+
+	return priv->frequency;
+}
+
+/************************************************************************************
+ * Name: amebad_spi_setmode
+ *
+ * Description:
+ *   Set the SPI mode.  see enum spi_mode_e mode for mode definitions
+ *
+ * Input Parameters:
+ *   dev  - Device-specific state data
+ *   mode - The SPI mode requested
+ *
+ * Returned Value:
+ *   Returns the actual frequency selected
+ *
+ ************************************************************************************/
+
+static void amebad_spi_setmode(FAR struct spi_dev_s *dev, enum spi_mode_e mode)
+{
+	FAR struct amebad_spidev_s *priv = (FAR struct amebad_spidev_s *)dev;
+
+	spiinfo("mode=%d\n", mode);
+
+	/* Has the mode changed? */
+
+	if (mode != priv->mode) {
+		/* Disable SPI if it is enabled */
+
+		priv->mode = mode;
+
+	}
+}
+
+/************************************************************************************
+ * Name: amebad_spi_setbits
+ *
+ * Description:
+ *   Set the number of bits per word.
+ *
+ * Input Parameters:
+ *   dev   - Device-specific state data
+ *   nbits - The number of bits requested
+ *
+ * Returned Value:
+ *   None
+ *
+ ************************************************************************************/
+
+static void amebad_spi_setbits(FAR struct spi_dev_s *dev, int nbits)
+{
+	FAR struct amebad_spidev_s *priv = (FAR struct amebad_spidev_s *)dev;
+
+	spiinfo("nbits=%d\n", nbits);
+
+	/* Has the number of bits changed? */
+
+	if (nbits != priv->nbits) {
+
+		if (nbits < 4 || nbits > 16) {
+			return;
+		}
+
+		/* Save the selection so the subsequence re-configurations will be faster */
+
+		priv->nbits = nbits;
+
+	}
+}
+
+/****************************************************************************
+ * Name: amebad_spi_hwfeatures
+ *
+ * Description:
+ *   Set hardware-specific feature flags.
+ *
+ * Input Parameters:
+ *   dev      - Device-specific state data
+ *   features - H/W feature flags
+ *
+ * Returned Value:
+ *   Zero (OK) if the selected H/W features are enabled; A negated errno
+ *   value if any H/W feature is not supportable.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SPI_HWFEATURES
+static int amebad_spi_hwfeatures(FAR struct spi_dev_s *dev,
+				amebad_spi_hwfeatures_t features)
+{
+#ifdef CONFIG_SPI_BITORDER
```

Comment:
I think we can reduce the code like below.
```
#ifdef CONFIG_SPI_BITORDER
...
#endif
return -ENOSYS;
```

Reply:
please check his comment.